consensus/ethash/ethash.go

```diff
@@ -320,15 +320,25 @@ func MakeDataset(block uint64, dir string) {
 	d.release()
 }
 
+// EthashConfig are the configuration parameters of the ethash.
+type EthashConfig struct {
+	CacheDir       string
+	CachesInMem    int
+	CachesOnDisk   int
+	DatasetDir     string
+	DatasetsInMem  int
+	DatasetsOnDisk int
+
+	// Miscellaneous options
+	PowFake   bool
+	PowTest   bool
+	PowShared bool
+}
+
 // Ethash is a consensus engine based on proot-of-work implementing the ethash
 // algorithm.
 type Ethash struct {
-	cachedir     string // Data directory to store the verification caches
-	cachesinmem  int    // Number of caches to keep in memory
-	cachesondisk int    // Number of caches to keep on disk
-	dagdir       string // Data directory to store full mining datasets
-	dagsinmem    int    // Number of mining datasets to keep in memory
-	dagsondisk   int    // Number of mining datasets to keep on disk
+	EthashConfig
```

Comment:
I don't really like when fields from multiple structs are mixed together. It would be cleaner to just have a private `config` field for `Config`.